sources/io/streams/sequence-stream.dylan

```diff
@@ -95,7 +95,8 @@ define method initialize
     stream.final-position := _start
   else
     stream.final-position := size(stream-sequence(stream))
-  end
+  end;
+  stream.stream-element-type := element-type(stream-sequence(stream));
 end method initialize;
```

Comment:
I'm not sure about this one. This seems to override the value given to the element-type: init-keyword. I found this in the docs:

"The element-type init-keyword specifies the type of the elements in the file named by filename. This allows file elements to be represented abstractly; for instance, contiguous elements could be treated as a single database record."

So maybe the intention was for the elements of the underlying sequence to be converted to the type given by the element-type: init-keyword?

Reply:
Interesting point. I assume [this](https://opendylan.org/documentation/library-reference/io/streams.html#options-when-creating-file-streams) is your reference?
It's odd because it talks about treating contiguous elements as a database record but goes on to list the possible element types as `<byte>`, `<byte-character>` and `<unicode-character>`.
Also, that reference is specific to file streams; there doesn't seem to be an element-type: keyword for sequence streams (https://opendylan.org/documentation/library-reference/io/streams.html#io:streams:[sequence-stream])
This makes sense because a file on disk is just an array of bytes whereas a sequence must have an element type (even if it's `<object>`), so the user would have to impose a type/format onto the former but not the latter.
What do you think?

Reply:
I failed to read the list of element types somehow. That makes me think that the bit about database records is just wrong (or at best misleading) and should be deleted.  Even if it was the original intention, I don't think functionality like that has any place in the streams code. Interpretation of the data should probably be handled at a higher level.  (Go doesn't seem to even do the byte -> text conversion for you; you can call string([]byte) if you want a string.)

I think when I was looking at this I saw that element-type: is specified by `<basic-stream>`, meaning that technically you could use it when creating a `<sequence-stream>`, but it probably would have no effect.

In other words, I think this change is fine. Perhaps remove that database verbiage from the docs while you're at it?